### PR TITLE
fix choose document search #2657

### LINKED ISSUE #2657: MyBible documents do not appear in search in "Choose Document"
**To Reproduce**
1. Go to "Choose Document"
2. Look for your installed MyBible module and you should find it
3. Now search for any terms in the name or description of the MyBible module
4. -> Does not show up

**Bug was found on AndBible version**
Tested only on 5.0.700-beta but I believe this bug will be in production as well

### PR DESCRIPTION
**Pull request by RuggMatt**
**Describe the pull request content**
MyBible documents now appear when using search on "Choose document" screen. 

Closes #2657 

**Screenshots**
Screenshot_20230904-182333

### CODE REVIEW

File: app/src/main/java/net/bible/android/view/activity/base/DocumentSelectionBase.kt [hunk 1]
```diff
@@ -427,11 +427,21 @@ abstract class DocumentSelectionBase(
                         val searchString = "${binding.freeTextSearch.text}*"
                         val osisIds = if(searchString.length < 3) null else dao.search(searchString).toSet()
 
+                        val test = { it: Book ->
+                            Log.d(TAG, "${it.name}")
```
Comment (RuggMatt) [inline app/src/main/java/net/bible/android/view/activity/base/DocumentSelectionBase.kt]:
```suggestion
```

File: app/src/main/java/net/bible/android/view/activity/base/DocumentSelectionBase.kt [hunk 2]
```diff
@@ -427,11 +427,20 @@ abstract class DocumentSelectionBase(
                         val searchString = "${binding.freeTextSearch.text}*"
                         val osisIds = if(searchString.length < 3) null else dao.search(searchString).toSet()
 
+                        val test = { it: Book ->
+                            // matches from list of osisIds, if any
+                            (osisIds?.contains(it.osisID) ?: (searchString.length < 3)) ||
```
Comment (tuomas2) [inline app/src/main/java/net/bible/android/view/activity/base/DocumentSelectionBase.kt:432]: instead of osisID should we look for initials?

Comment (RuggMatt) [reply app/src/main/java/net/bible/android/view/activity/base/DocumentSelectionBase.kt:432]: I think that makes more sense. I was just using what was there before.

**Comment by tuomas2** [comment]
I chose a bit different approach to do this, similar in Download documents activity, we index entities in FTS4 table. 1db398d0c6d19668fc8e0541ee794c9a2163e7ff

**Pull request closed without merging**